File: Trees/InvertBT.py
#https://leetcode.com/problems/invert-binary-tree/description/



class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

#DFS - Recursion preorder
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None        
        #swap children
        temp = root.left
        root.left = root.right
        root.right = temp
        self.invertTree(root.left)
        self.invertTree(root.right)
        return root
    
# Recursion
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None
        left = self.invertTree(root.left)
        right = self.invertTree(root.right)
        root.left = right
        root.right = left
        return root 
    

        

File: Trees/InvertBT.java
package Trees;
//https://leetcode.com/problems/invert-binary-tree/description/

class TreeNode{
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(){}

    TreeNode(int val){
        this.val = val;
    }
    TreeNode(int val, TreeNode left, TreeNode right){
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution{
    public TreeNode invertTree(TreeNode root){
        if (root == null)
            return null;
        TreeNode left = this.invertTree(root.left);
        TreeNode right = this.invertTree(root.right);
        root.left = right;
        root.right = left;
        return root;
    }
}